inst/examples/_render.R

```diff
@@ -12,40 +9,36 @@ owd = setwd(dirname(src))
 
 # provide default formats if necessary
 if (length(formats) == 0) formats = c(
-  'bookdown::pdf_book', 'bookdown::epub_book', 'bookdown::gitbook'
+  'bookdown::pdf_book', 'bookdown::epub_book', 'bookdown::gitbook', 'bookdown::bs4_book'
 )
 # render the book to all formats unless they are specified via command-line args
+message(">> Building Books:")
 for (fmt in formats) {
-  cmd = sprintf("bookdown::render_book('index.Rmd', '%s', quiet = %s)", fmt, quiet)
+  message("  -- Building format: ", fmt)
+  # change output_dir for bs4_book as we don't want to publish it
+  output_dir = if (fmt == 'bookdown::bs4_book') "'_bs4_book'" else "NULL"
+  cmd = sprintf("bookdown::render_book('index.Rmd', '%s', quiet = %s, output_dir = %s)", fmt, quiet, output_dir)
   res = xfun::Rscript(c('-e', shQuote(cmd)))
   if (res != 0) stop('Failed to compile the book to ', fmt)
-  if (!travis && fmt == 'bookdown::epub_book')
-    bookdown::calibre('_book/bookdown.epub', 'mobi')
+  if (fmt == 'bookdown::epub_book') bookdown::calibre('_book/bookdown.epub', 'mobi')
 }
 unlink('bookdown.log')
 
 # tweak the generated html files
-r = '<body onload="window.location = \'https://bookdown.org/yihui\'+location.pathname">'
+message(">> Tweaking HTML gitbook")
 for (f in list.files('_book', '[.]html$', full.names = TRUE)) {
   x = readLines(f)
   if (length(i <- grep('^\\s*<body>\\s*$', x)) == 0) next
-  # patch HTML files in gh-pages if built on Travis, to redirect to bookdown.org
-  if (travis) x[i[1]] = r
   i = grep('<i class="fa fa-circle-o-notch fa-spin"></i><a href="./">.+</a>', x)[1]
   # shorter title on the toolbar
   if (!is.na(i)) x[i] = gsub('bookdown: ', '', x[i], fixed = TRUE)
-  i = c(
-    grep('&lt;bytecode: 0x[0-9a-f]+&gt;$', x),
-    grep('^\\s*<meta name="generator" content="bookdown [.0-9]+ and GitBook [.0-9]+" />$', x),
-    grep('^<meta name="date" content="[-0-9]+" />$', x)
-  )
-  if (travis && length(i)) x = x[-i]
```

Comment:
Oh I suddenly recall the reason for publishing to `gh-pages`: I wanted to check the differences in HTML when I switch to a newer version of Pandoc, e.g., https://groups.google.com/g/pandoc-discuss/c/1bwAre8fG6E/m/k4vM455kCQAJ If you feel that's useful, perhaps we need to keep these tweaks and keep pushing to `gh-pages`.

Reply:
Oh I see. So the that is why when looking a the webpage `github.io` url, it would redirect to bookdown, but you still could look at the gh-pages branch to see the change. 

That sounds useful. But we currently are build the gh-page branch on every commit to master and I did not see why. It seemed too much to run a CI for nothing.

So now it is not nothing as I understand. But I know I have never looked at the branch `gh-pages` 😅 
Would it be useful to have something we run manually when we want to test a Pandoc version ? 
Or do you look at this branch sometime ? 

Currently, we build the gitbook book on PR, but we can't see the change. Only the book folder can be downloaded on error in the build. 

I am ok to keep it as I see how it can be useful, but maybe I should do it in GHA too to move from travis completly. And maybe we don't need to run it on each commit ? Manual trigger maybe ?

Reply:
> Would it be useful to have something we run manually when we want to test a Pandoc version ?

Yes, if we have that, the `gh-pages` branch will be useless and we can remove it.